### ISSUE TITLE
Support special behavior for upgrading Squirrel applications

### ISSUE THREAD

**Issue by jedieaston** [microsoft/winget-cli#1748]
### Description of the new feature / enhancement

While investigating #1747 I found out that the [Squirrel installer](https://github.com/Squirrel/Squirrel.Windows) has a special way to do upgrades. Namely, it uses a Update.exe file inside the application's install directory (usually User\AppData\Local) with some options to execute it. WinGet should support running this executable with the correct options to upgrade in lieu of running the installer over again, which can in certain cases (again, #1747) cause unexpected behavior.

It's worth noting that Teams is one of the applications that uses the Squirrel installer[^1], so this may be part of the answer to that problem.

[^1]:  If you want to know where the madness with the machine wide installer came from, check it out [here](https://github.com/Squirrel/Squirrel.Windows/blob/develop/docs/using/machine-wide-installs.md)

### Proposed technical implementation details

A new InstallerType (`squirrel`) that has defaults for silent install arguments (`/s`) and for upgrading using the Update.exe file instead of the installer.

**Comment by denelon** [comment]
Related to:

* #372
* #418

**Comment by xarthurx** [comment]
As of today, the issue for the `Fork` software is still unresolved.
Every time I got an update, the info is lost.

**Comment by xarthurx** [comment]
Another half year, and as of today, the issue for the `Fork` software is still unresolved.
Every time I got an update, the info is lost.

**Comment by xarthurx** [comment]
@denelon Is there anyone working on this issue?

**Comment by denelon** [comment]
@xarthurx no, this is a fairly large feature. We have to fundamentally change the way we're calling installers to watch an entire process tree.

Related to:
* #372 
* #418
